internal/internal_task_handlers.go

```diff
@@ -635,13 +640,14 @@ func (wth *workflowTaskHandlerImpl) getOrCreateWorkflowContext(
 
 		if wth.cache.MaxWorkflowCacheSize() > 0 && task.Query == nil {
 			workflowContext, _ = wth.cache.putWorkflowContext(runID, workflowContext)
+			cached = true
 		}
 		workflowContext.Lock()
 	}
 
```

Comment:
Looks like maybe you could just add `cached := workflowContext != nil` here rather than needing to set it in two of the branches above?

Reply:
GitHub does a bad job of indenting tab-based languages. Sometimes `workflowContext` is created without being cached. I only set `cached` as true if it was successfully retrieved or successfully put in cache, not if it just happens to be present.

Reply:
Got it